### PR TITLE
Support ghc 9.2.1

### PR DESCRIPTION
**Pull request by shayne-fletcher**
- Add support for ghc flavor ghc-9.2.1
- Rebase this PR on master after https://github.com/digital-asset/ghc-lib/pull/289 lands before landing;
- Sync to `ce706faeef3964116c6e1dd0e6ae2f2e77fde57d` @ https://gitlab.haskell.org/ghc/ghc.git

### CODE REVIEW

File: examples/mini-compile/src/Main.hs [hunk 1]
```diff
@@ -18,7 +20,7 @@ module Main (main) where
 
 import "ghc-lib" GHC
 import "ghc-lib" Paths_ghc_lib
-#if defined (GHC_MASTER) || defined (GHC_901)
+#if defined (GHC_MASTER) ||  defined (GHC_921) || defined (GHC_901)
```
Comment (cocreature) [inline examples/mini-compile/src/Main.hs:23]: we might want to think about a `MIN_version` like pragma for this. This seems to be getting a bit out of hand.

**Review by cocreature — COMMENTED** [review]
lgtm but let’s merge #289  first

**Comment by shayne-fletcher** [comment]
Rebased on https://github.com/shayne-fletcher/ghc-lib/pull/52. After that lands, rebase once again on master, then land.

**Comment by shayne-fletcher** [comment]
rebased on master and hopefully ready to go

**Review by cocreature — APPROVED** [review]

**Pull request merged**